src/NumericalAlgorithms/LinearOperators/PartialDerivatives.hpp

```diff
@@ -319,6 +411,43 @@ struct DerivCompute
   using argument_tags = tmpl::list<VariablesTag, MeshTag, InverseJacobianTag>;
 };
 
+template <typename VariablesTag, typename MeshTag, typename InverseJacobianTag,
+          typename DerivTags, typename InertialCoordsTag>
+struct DerivCompute
+    : db::add_tag_prefix<
+          deriv, ::Tags::Variables<DerivTags>,
+          tmpl::size_t<
+              tmpl::back<typename InverseJacobianTag::type::index_list>::dim>,
+          typename tmpl::back<
+              typename InverseJacobianTag::type::index_list>::Frame>,
+      db::ComputeTag {
+ private:
+  using inv_jac_indices = typename InverseJacobianTag::type::index_list;
+  static constexpr auto Dim = tmpl::back<inv_jac_indices>::dim;
+  using deriv_frame = typename tmpl::back<inv_jac_indices>::Frame;
+
+ public:
+  using base = db::add_tag_prefix<
+      deriv, ::Tags::Variables<DerivTags>,
+      tmpl::size_t<
+          tmpl::back<typename InverseJacobianTag::type::index_list>::dim>,
+      typename tmpl::back<
+          typename InverseJacobianTag::type::index_list>::Frame>;
+  using return_type = typename base::type;
+  //NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
+  static constexpr void (*function)(
```

Comment:
Re your PR comment, you can do a function instead of a function pointer, e.g.
```cpp
static void function(gsl::not_null<return_type*>, const typename VariablesTag::type&,
      const Mesh<Dim>&,
      const InverseJacobian<DataVector, Dim, Frame::ElementLogical,
                            deriv_frame>&,
      const tnsr::I<DataVector, Dim, Frame::Inertial>&) {

}

static void function(gsl::not_null<return_type*>, const typename VariablesTag::type&,
      const Mesh<Dim>&,
      const InverseJacobian<DataVector, Dim, Frame::ElementLogical,
                            deriv_frame>&) {

}
```
and then you have overloads and only change the argument tags based on whether or not the `InertialCoordsTag` is set